index.d.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * Handler signature for when an extension updates.
+ */
+export interface ExtensionUpdateHandler {
+	(savedOptions: Record<string,string|boolean>, defaults: Record<string, string|boolean>): void;
+}
+
+/**
+ * @example
+ * {
+ * 	defaults: { // recommended
+ * 		color: 'blue'
+ *	},
+ * 	migrations: [ // optional
+ * 		savedOptions => {
+ * 			if(savedOptions.oldStuff) {
+ * 				delete savedOptions.oldStuff
+ * 			}
+ * 		}
+ * 	],
+ * }
+ */
+export interface DefineOptions {
+	defaults: Record<string, string|boolean>;
+	/**
+	 * A list of functions to call when the extension is updated.
+	 */
+	migrations: ExtensionUpdateHandler[];
+}
+
+/**
+ * Listens to any field that triggers `input` or `change` events.
+ * Option names are set via the fields' `name` attribute.
+ * Checkboxes are stored as `true`/`false`; other fields are stored as strings.
+ */
+export default class OptionsSync {
+	constructor(config: {storageName: string });
+
+	static migrations: {
+		/**
+		 * Helper method that removes any option that isn't defined in the defaults. It's useful to avoid leaving old options taking up space.
+		 */
+		removeUnused: () => void;
+	}
+
+	/**
+	 * Retrieves all the options stored.
+	 *
+	 * @returns Promise that will resolve with **all** the options stored, as an object.
+	 *
+	 * @example
+	 * new OptionsSync().getAll().then(options => {
+	 * 	console.log('The user’s options are', options);
+	 * 	if(options.color) {
+	 * 		document.body.style.color = color;
+	 * 	}
+	 * });
+	 * });
+	 */
+	getAll: <T>() => Promise<T>;
+
+	/**
+	 * Overrides **all** the options stored with your `options`
+	 *
+	 * @param {Record<string, string}|boolean>} A map of default options as strings or booleans. The keys will have to match the form fields' `name` attributes.
+	 */
+	setAll: (newOptions: Record<string, string|boolean>) => Promise<void>;
+
+	/**
+	 * Merges new options with the existing stored options.
```

Comment:
There was no doc for this method so I winged it based on what the code does. Happy to update the wording if not to your liking.